contrib/install_db4.sh

```diff
@@ -68,8 +68,8 @@ tar -xzvf ${BDB_VERSION}.tar.gz -C "$BDB_PREFIX"
 cd "${BDB_PREFIX}/${BDB_VERSION}/"
 
 # Apply a patch necessary when building with clang and c++11 (see https://community.oracle.com/thread/3952592)
-CLANG_CXX11_PATCH_URL='https://gist.githubusercontent.com/LnL7/5153b251fd525fe15de69b67e63a6075/raw/7778e9364679093a32dec2908656738e16b6bdcb/clang.patch'
-CLANG_CXX11_PATCH_HASH='7a9a47b03fd5fb93a16ef42235fa9512db9b0829cfc3bdf90edd3ec1f44d637c'
+CLANG_CXX11_PATCH_URL='https://raw.githubusercontent.com/bitcoin/bitcoin/master/depends/patches/bdb/clang_cxx_11.patch'
```

Comment:
if it's in the same repository, does it need a download at all?

Reply:
I see this script as a standalone script which installs libdb4.8 (Berkeley DB) without the need to be in the root of the bitcoin repository. One can just grab this script and run it anywhere on the machine, export BDB_PREFIX and then use the right parameters (BDB_LIBS= and BDB_CFLAGS) on configure

Reply:
Yes, good point. Well I guess it could at least *try* finding the local patch, and resort to download if it doesn't exist.

Reply:
There is convenience for running curl from anywhere, yet are there other tradeoffs in depending on githubusercontent.com?

Reply:
Possibly an even better solution: embed the patch in the script, with a doc string.

It's more maintenance work *when* the patch needs to change (say, for compatibility with new compilers), but it is a small patch after all and it keeps the script more self-contained.

Reply:
Good idea. Also, this script will probably go in one or two major releases when we no longer require bdb

Reply:
Implemented you idea @laanwj